ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/resources/KsqlResourceTest.java

```diff
@@ -689,6 +695,67 @@ public void shouldNotIncludeSslPropertiesInListPropertiesOutput() {
         not(hasItems(KsqlConfig.SSL_CONFIG_NAMES.toArray(new String[0]))));
   }
 
+  @Test
+  public void shouldNotWaitIfNoCommandTopicOffsetSpecified() {
+    // Given:
+    final String sql = "LIST REGISTERED TOPICS;";
+    final KsqlRequest request = new KsqlRequest(sql, Collections.emptyMap(), null);
+
+    EasyMock.expect(commandStore.getConsumerPositionFuture(EasyMock.anyLong()))
+        .andThrow(new AssertionFailedError()).anyTimes();
+    EasyMock.replay(commandStore);
+
+    // When:
+    makeSingleRequest(request, KsqlTopicsList.class);
+
+    // Then:
+    EasyMock.verify(commandStore);
+  }
+
+  @Test
+  public void shouldNotWaitIfCommandTopicOffsetReached()
+      throws InterruptedException, ExecutionException, TimeoutException {
+    // Given:
+    final String sql = "LIST REGISTERED TOPICS;";
+    final KsqlRequest request = new KsqlRequest(sql, Collections.emptyMap(), 2L);
+
+    final CompletableFuture<Void> future = EasyMock.mock(CompletableFuture.class);
+    EasyMock.expect(future.get(EasyMock.anyLong(), EasyMock.anyObject())).andReturn(null);
+    EasyMock.expect(commandStore.getConsumerPositionFuture(EasyMock.anyLong()))
+        .andReturn(future);
+    EasyMock.replay(future, commandStore);
+
+    // When:
+    makeSingleRequest(request, KsqlTopicsList.class);
+
+    // Then:
+    EasyMock.verify(future, commandStore);
+  }
+
+  @Test
+  public void shouldReturn5xxIfTimeoutWhileWaitingForCommandTopicOffset()
+      throws InterruptedException, ExecutionException, TimeoutException {
+    // Given:
+    final String sql = "LIST REGISTERED TOPICS;";
+    final KsqlRequest request = new KsqlRequest(sql, Collections.emptyMap(), 2L);
+
+    final CompletableFuture<Void> future = EasyMock.mock(CompletableFuture.class);
+    EasyMock.expect(future.get(EasyMock.anyLong(), EasyMock.anyObject()))
+        .andThrow(new TimeoutException());
+    EasyMock.expect(commandStore.getConsumerPositionFuture(EasyMock.anyLong()))
+        .andReturn(future);
+    EasyMock.replay(future, commandStore);
+
+    // When:
+    final KsqlErrorMessage result = makeFailingRequest(request, Code.INTERNAL_SERVER_ERROR);
```

Comment:
As mentioned already - a timeout shouldn't result in an server error status code!